src/hotspot/share/runtime/synchronizer.cpp

```diff
@@ -444,8 +446,8 @@ bool ObjectSynchronizer::quick_enter(oop obj, JavaThread* current,
 }
 
 // Handle notifications when synchronizing on value based classes
-void ObjectSynchronizer::handle_sync_on_value_based_class(Handle obj, JavaThread* current) {
-  frame last_frame = current->last_frame();
+void ObjectSynchronizer::handle_sync_on_value_based_class(Handle obj, JavaThread* locking_thread) {
```

Comment:
Can we add a comment (or even repeat the assertion) that locking thread is either the current thread or a suspended thread, please. This method performs a  number of actions that are not safe to make on an arbitrary thread.

Reply:
I breezed right by this function... where did that come from?

Reply:
Sorry, rhetorical question. I can figure out where it came from (and how I missed seeing it).